api/v1beta1/nutanixcluster_types.go

```diff
@@ -64,7 +64,17 @@ type NutanixClusterSpec struct {
 	// +listType=map
 	// +listMapKey=name
 	// +optional
-	FailureDomains []NutanixFailureDomain `json:"failureDomains"`
+	//
+	// Deprecated: This field is replaced by the field controlPlaneFailureDomains and will be removed in the next apiVersion.
+	//
+	FailureDomains []NutanixFailureDomainConfig `json:"failureDomains,omitempty"`
+
+	// controlPlaneFailureDomains configures references to the NutanixFailureDomain objects
+	// that the cluster uses to deploy its control-plane machines.
+	// +listType=map
+	// +listMapKey=name
```

Comment:
I dont think we need these any more. local object reference is only names

Reply:
Using the map type with key of "name" can make sure there is no duplicate failureDomain configuration in the list.

Reply:
if you want no duplicate names use this

```
// +kubebuilder:validation:UniqueItems
```
https://book.kubebuilder.io/reference/markers/crd-validation

Reply:
i created a PR against this https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/pull/531 to show that it does work. I prefer that we use this simpler validation